math-scala/src/main/scala/org/apache/mahout/math/decompositions/DSSVD.scala

```diff
@@ -22,13 +22,18 @@ object DSSVD {
    * @return (U,V,s). Note that U, V are non-checkpointed matrices (i.e. one needs to actually use them
    *         e.g. save them to hdfs in order to trigger their computation.
    */
-  def dssvd[K](drmA: DrmLike[K], k: Int, p: Int = 15, q: Int = 0):
+  def dssvd[K](drmA: DrmLike[K],
+               k: Int,
+               p: Int = 15,
+               q: Int = 0,
+               cacheHint: CacheHint.CacheHint = CacheHint.MEMORY_ONLY):
```

Comment:
Not sure what we should make the default `CacheHint` here and other places, e.g. #276.

for `drmDfsRead(...)` we use `CacheHint.NONE` but that may be something that we should let the user supply as well.  Most parallaization methods use `CacheHint.NONE`.  I would think that if we have anything hardcoded or as a default, we should use `CacheHint.MEMORY_AND_DISK` since we have no idea of the size of the user's data..  but I suppose that is making Assumptions as well.